src/utils/validation.ts

```diff
@@ -16,31 +17,33 @@ export function validatePositive(value: string, constraint = 0): true | string {
   return Number(value) > constraint || 'Invalid amount'
 }
 
-export const resolverFactory = (type: string, validationSchema: ObjectSchema<unknown>) => (
-  data: FormData,
-): { values: ValidationResult; errors: ValidationError | {} } => {
-  const castedData =
+export const resolverFactory = <D, CustomContext = {}>(type: string, validationSchema: ObjectSchema<unknown>) => (
+  data: D,
+): ReturnType<ValidationResolver<D, CustomContext>> => {
+  const castedData: D =
     type === 'number'
       ? Object.keys(data).reduce((acc, key) => {
           const oldValue = data[key]
           const castedValue = Number(oldValue)
-          const castedObj = { ...acc, [key]: castedValue }
+          const castedObj: D = { ...acc, [key]: castedValue }
 
           return castedObj
-        }, {})
+        }, {} as D)
```

Comment:
I'm not sure of it but try `Object.keys(data).reduce<D>` instead of `as D`
If doesn't help, then whatever